src/io/jesi/backpack/macros.cljc

```diff
@@ -220,3 +222,65 @@ A single default expression can follow the clauses, and its value will be return
    `(def ~(private symbol) ~init))
   ([symbol doc-string init]
    `(def ~(private symbol) ~doc-string ~init)))
+
+(defmacro with-open
+  "bindings => [sym init ...]
+
+  Evaluates body in a try expression with symbols bound to the values
+  of the inits, and a finally clause that calls (bp/close sym) on each
+  sym in reverse order."
+  [bindings & body]
+  {:pre [(vector? bindings)
+         (even? (count bindings))]}
+  (cond
+    (zero? (count bindings))
+    `(do ~@body)
+
+    (symbol? (bindings 0))
+    `(let ~(subvec bindings 0 2)
+       (try
+         (with-open ~(subvec bindings 2)
+           ~@body)
+         (finally
+           (close ~(bindings 0)))))
+
+    :else
+    (throw (#?(:clj IllegalArgumentException. :cljs js/Error.) "with-open only allows Symbols in bindings"))))
+
+(defmacro with-open-> [x & forms]
+  `(with-open [v# ~x]
+     (-> v#
+         ~@forms)))
+
+(defmacro assoc-nx
+  "Lazily assocs each kv iff its key doesn't already exist.
```

Comment:
```suggestion
  "Lazily assocs each kv if its key doesn't already exist.
```

Reply:
That what was is the `assoc-nx`'s docstring. I think "iff" mean "if and only if" https://en.wikipedia.org/wiki/If_and_only_if
I'm happy to change it

Reply:
Nah, that's fine.